Source/WebKit/Scripts/webkit/tests/GeneratedWebKitSecureCoding.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2022-2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1.  Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ * 2.  Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if PLATFORM(COCOA)
+#include "CoreIPCTypes.h"
+
+OBJC_CLASS NSSomeFoundationType;
+#if ENABLE(DATA_DETECTION)
+OBJC_CLASS DDScannerResult;
+#endif
+
+namespace WebKit {
+
+class CoreIPCNSSomeFoundationType {
+public:
+    CoreIPCNSSomeFoundationType(NSSomeFoundationType *);
+    CoreIPCNSSomeFoundationType(RetainPtr<NSSomeFoundationType> object)
+        : CoreIPCNSSomeFoundationType(object.get())
+    {
+    }
+
+    static bool isValidDictionary(CoreIPCDictionary&);
+    RetainPtr<id> toID() const;
+
+private:
+    friend struct IPC::ArgumentCoder<CoreIPCNSSomeFoundationType, void>;
+
+    CoreIPCNSSomeFoundationType(CoreIPCDictionary&& propertyList)
+        : m_propertyList(WTFMove(propertyList))
+    {
+    }
+
+    CoreIPCDictionary m_propertyList;
```

Comment:
Maybe not for this type, but many types will need this to be a std::optional to preserve nullness.  That mistake has bitten me several times this year.

Reply:
I don't forsee that in this case:
If we're wrapping an object that actually exists, then by definition it cannot be null...
And then our contract for the webkit property list model is there will always be a dictionary.
The dictionary might be 100% optional keys and therefore empty, but the dictionary itself will be present.

If a situation comes up that breaks that assumption, we can change the generator then. But for now my approach is to only add the required features as they come up.